ocs_ci/utility/aws.py

```diff
@@ -1994,17 +1994,30 @@ def delete_objects_in_bucket(self, bucket):
             bucket (str): Name of the bucket to delete objects
 
         """
-        # List all objects within the bucket
-        response = self.s3_client.list_objects_v2(Bucket=bucket)
-
-        # Delete each object within the bucket
-        if "Contents" in response:
-            for obj in response["Contents"]:
-                object_key = obj["Key"]
-                self.s3_client.delete_object(Bucket=bucket, Key=object_key)
-                logger.info(f"Deleted object: {object_key}")
-        else:
-            logger.info(f"No objects found in bucket {bucket}")
+        try:
+            # List all objects within the bucket
+            response = self.s3_client.list_objects_v2(Bucket=bucket)
+
+            # Delete each object within the bucket
+            if "Contents" in response:
+                for obj in response["Contents"]:
+                    object_key = obj["Key"]
+                    self.s3_client.delete_object(Bucket=bucket, Key=object_key)
+                    logger.info(f"Deleted object: {object_key}")
+            else:
+                logger.info(f"No objects found in bucket {bucket}")
+
+            # Delete each version objects
+            versioning = self.s3_client.get_bucket_versioning(Bucket=bucket)
+            s3_resource = boto3.resource("s3")
+            if versioning.get("Status") == "Enabled":
+                version_objs = s3_resource.Bucket(bucket).object_versions.all()
+                for version_obj in version_objs:
+                    version_obj.delete()
+        except Exception as e:
+            logger.info(f"Cant delete objects {e}")
```

Comment:
change info to warn/error. Also add Return to Args section.

is there any reason to suppress the exception because we are not dealing with failures in the function where we are calling this https://github.com/red-hat-storage/ocs-ci/pull/9861/files#diff-44938617241aefc7f12e348ee2af0cf229899d035156d229c9cf2b33f0b3b81eL2009

Reply:
If you fail to delete all the objects in the bucket, you cannot delete the bucket. so we retun false and we dont try to delete the bucket.